### ISSUE TITLE
Unified Focal Loss

### ISSUE THREAD

**Issue by danieltudosiu** [Project-MONAI/MONAI#4413]
**Is your feature request related to a problem? Please describe.**
Having the best loss available.

**Describe the solution you'd like**
Implement the Unified Focal Loss

**Additional context**
The Unified Focal Loss has been shown to be constantly better than quite a few other alternatives that are implemented din MONAI.

Paper: https://www.sciencedirect.com/science/article/pii/S0895611121001750
Original Tensorflow Code: https://github.com/mlyg/unified-focal-loss/blob/main/loss_functions.py

**Comment by JohnMasoner** [comment]
hi @Nic-Ma ,

I think it's an exciting idea. 

I am trying to go code with this proposal. but i'm a first-time contributor, I had to spend some time reading CONTRIBUTING.md.

Mason

**Comment by danieltudosiu** [comment]
Hi @JohnMasoner,

Internally a coworker of mine coded the Symmetrical Unified Focal Loss like this:
```
import torch
from monai.losses import TverskyLoss, FocalLoss

class Unified_FL(torch.nn.Module):
    def __init__(self, n_classes, delta = 0.6, gamma = 0.5, lambda_ = 0.5):
        super(Unified_FL, self).__init__()
        self.N = n_classes
        self.delta = delta
        self.gamma = gamma
        self.lambda_ = lambda_
        self.tversky = TverskyLoss(include_background=True, alpha = self.delta, beta = (1-self.delta),
                                   reduction="mean")
        self.focal = FocalLoss(include_background=True, gamma = 1-self.gamma, reduction="mean")
    def forward(self, x, y):
        return self.lambda_ * self.delta * self.focal(x, y) + (1 - self.lambda_) * self.tversky(x, y)
```

She said that for the Asymmetrical one the TverskyLoss needs to be modified.

Hope it helps ;) 

And thanks for picking it up <3

**Comment by pedrohbd** [comment]
I noticed the code for this loss is available in MONAI repositories, but it seems to work just for binary segmentation. I assume this is the reason it is not available in the documentation yet. If someone is willing to take the task, the author of the loss has provided a multiclass code in an issue: https://github.com/mlyg/unified-focal-loss/issues/8. He didn't update his repository to add this multiclass version though...

**Comment by JohnMasoner** [comment]
> I noticed the code for this loss is available in MONAI repositories, but it seems to work just for binary segmentation. I assume this is the reason it is not available in the documentation yet. If someone is willing to take the task, the author of the loss has provided a multiclass code in an issue: [mlyg/unified-focal-loss#8](https://github.com/mlyg/unified-focal-loss/issues/8). He didn't update his repository to add this multiclass version though...

Hey @pedrohbd ,

Thanks for your advice and focus on the work.
Actually, this is a TODO thing. I will get this done as quickly as possible. (Of course if you have a suitable candidate you can also recommend to us.)

Mason.
